flake.nix

```diff
@@ -97,6 +99,19 @@
           '';
         };
 
+        devShells.openssl3 = devShells.default.overrideAttrs
+          (finalAttrs: previousAttrs: {
+            # Re-include cmake to update the environment with a new libcrypto.
+            buildInputs = [ pkgs.cmake openssl_3_0 ];
+            S2N_LIBCRYPTO = "openssl-3.0";
+            shellHook = ''
+              echo Setting up $S2N_LIBCRYPTO enviornment from flake.nix...
+              export PATH=${openssl_1_1_1}/bin:${gnutls-3-7}/bin:$PATH
+              export PS1="[nix $S2N_LIBCRYPTO] $PS1"
+              source ${writeScript ./nix/shell.sh}
+            '';
+          });
+
         # Used to backup the devShell to s3 for caching.
         packages.devShell = devShells.default.inputDerivation;
         packages.default = packages.s2n-tls;
```

Comment:
on line 58 can we:
   buildInputs = [ pkgs.openssl-openss-3.0 ];


and can we remove `propagatedBuildInputs = [ pkgs.openssl ];`?

Reply:
> propagatedBuildInputs is intended for packages that are required for both building and running of the package.

and this should also get the specific version if buildInputs does.

Reply:
can we drop buildInputs and just use propagatedBuildInputs then?

Reply:
Tested dropping buildInputs, no it appears they both need to be present.